libcaf_core/caf/event_based_mail.test.cpp

```diff
@@ -243,6 +244,83 @@ TEST("send request message") {
       expect<exit_msg>().to(dummy);
     }
   }
+  SECTION("using .to_observable for the response") {
+    SECTION("valid response") {
```

Comment:
We have "valid response" nested twice, maybe this description would be a better match with the below section.
```suggestion
    SECTION("response from dynamically typed receiver") {
```